### PR TITLE
Merge/merge old repo

### PR DESCRIPTION
**Pull request by dylanratcliffe**

### CODE REVIEW

**Comment by danzilio** [comment]
Holy shit. This is awesome! I'll definitely merge this in, I just have a few housekeeping things that I'd like to see first. It looks like there's a merge conflict in README.md; could you take a look at that? There's also a couple errant symlinks in `spec/fixtures/modules` that should be removed and the files at `spec/defines/virtual_machine_spec.rb` and `spec/classes/init_spec.rb` should be removed as they're effectively empty. Also, for the sake of our git history, could you squash this into one commit?

Thanks!!

File: spec/acceptance/03_virtualbox_type_spec.rb [hunk 1]
```diff
@@ -0,0 +1,355 @@
...
+      end
+    end
+
+    it 'should have the correct description' do
+      command('VBoxManage showvminfo computer --machinereadable') do
+        its(:exit_status) { should eq 0 }
+        its(:stdout) { should match /description="Managed by puppet"/ }
+      end
+    end
+
```
Comment (danzilio) [inline spec/acceptance/03_virtualbox_type_spec.rb]: Also, I think these 'should have the correct *' could be compressed into one example:

```ruby
it 'should have the correct settings' do
  command('VBoxManage showvminfo computer --machinereadable') do
    its(:exit_status) { should eq 0 }
    its(:stdout) do
      should match /description="Managed by puppet"/
      should match /type=Other\/Unknown/
      should match /memory=512/
      ...and so on...
    end
  end
end

This way we only have to execute `VBoxManage showvminfo computer --machinereadable` once!

Comment (dylanratcliffe) [inline spec/acceptance/03_virtualbox_type_spec.rb]: Good idea, will do

**Comment by dylanratcliffe** [comment]
Okay that should all be good to go, I think...

**Comment by dylanratcliffe** [comment]
Actually wait I have failing beaker tests, i'll fix them

**Comment by dylanratcliffe** [comment]
Okay so turns out this behaves differently based on your OS, some need explicit setting of storage media. I'll close this while I work on it

**Pull request closed without merging**